kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -98,6 +107,108 @@ public static Map<String, Map<String, Variant<?>>> buildSettings(NetworkProperti
+        }
+
+        return settings;
+    }
+
+    private static void build8021xTunneledTls(NetworkProperties props, String deviceId,
+            Map<String, Variant<?>> settings) {
+        settings.put("eap", new Variant<>(new String[] { NM8021xEAP.ttls.name() }));
+        build8021xOptionalCaCertAndAnonIdentity(props, deviceId, settings);
+    }
+
+    private static void build8021xProtectedEap(NetworkProperties props, String deviceId,
+            Map<String, Variant<?>> settings) {
+        settings.put("eap", new Variant<>(new String[] { NM8021xEAP.peap.name() }));
+        build8021xOptionalCaCertAndAnonIdentity(props, deviceId, settings);
+    }
+
+    private static void build8021xTls(NetworkProperties props, String deviceId, Map<String, Variant<?>> settings) {
+        settings.put("eap", new Variant<>(new String[] { NM8021xEAP.tls.name() }));
+        build8021xOptionalCaCertAndAnonIdentity(props, deviceId, settings);
+
+        String identity = props.get(String.class, "net.interface.%s.config.802-1x.identity", deviceId);
+        settings.put("identity", new Variant<>(identity));
+
+        String clientCert = props.get(String.class, "net.interface.%s.config.802-1x.client-cert", deviceId);
+        settings.put("client-cert", new Variant<>(clientCert.getBytes(StandardCharsets.UTF_8)));
+
+        String privateKey = props.get(String.class, "net.interface.%s.config.802-1x.private-key", deviceId);
+        settings.put("private-key", new Variant<>(privateKey.getBytes(StandardCharsets.UTF_8)));
+
+        String privateKeyPassword = props
+                .get(Password.class, "net.interface.%s.config.802-1x.private-key-password", deviceId).toString();
+        settings.put("private-key-password", new Variant<>(privateKeyPassword));
+
+    }
+
+    private static void build8021xOptionalCaCertAndAnonIdentity(NetworkProperties props, String deviceId,
+            Map<String, Variant<?>> settings) {
+        Optional<String> anonymousIdentity = props.getOpt(String.class,
+                "net.interface.%s.config.802-1x.anonymous-identity", deviceId);
+        if (anonymousIdentity.isPresent()) {
+            settings.put("anonymous-identity", new Variant<>(anonymousIdentity.get()));
+        }
+
+        Optional<String> caCert = props.getOpt(String.class, "net.interface.%s.config.802-1x.ca-cert", deviceId);
+        if (caCert.isPresent()) {
+            settings.put("ca-cert", new Variant<>(caCert.get().getBytes(StandardCharsets.UTF_8)));
+        }
+
+        Optional<Password> caCertPassword = props.getOpt(Password.class,
+                "net.interface.%s.config.802-1x.ca-cert-password", deviceId);
+        if (caCertPassword.isPresent()) {
+            settings.put("ca-cert-password", new Variant<>(caCertPassword.get().toString()));
+        }
+    }
+
+    private static void build8021xMschapV2(NetworkProperties props, String deviceId, Map<String, Variant<?>> settings) {
```

Comment:
Rename this method for consistency. We reserved `build*` for the top level public methods, `create*` for the private ones called by the `build*` ones.

```suggestion
    private static void create8021xMschapV2(NetworkProperties props, String deviceId, Map<String, Variant<?>> settings) {
```